### PR TITLE
[#720] [TECH] Ajout d'un script de suppression d'évaluation (US-1174).

### PR DESCRIPTION
**Pull request by Akhilian**

### CODE REVIEW

**Comment by Akhilian** [comment]
[ETQ PixMaster, JV supprimer un assessment pour plusieurs utilisateurs](https://trello.com/c/dVX3hDXa/1174-etq-pixmaster-jv-supprimer-un-assessment-pour-plusieurs-utilisateurs)

File: api/scripts/delete-assessment.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env node
+/* eslint no-console: ["off"] */
+const { Client } = require('pg');
+
+function initialize() {
+  const client = new Client(process.env.DATABASE_URL);
+  client.connect();
+
+  client.logged_query = function(query) {
```
Comment (mlaurent) [inline api/scripts/delete-assessment.js]: Je n'avais pas bien compris que logged_query exécutait la query ... ne peut-on pas renommer en "query_and_log" ?

**Review by mlaurent — CHANGES_REQUESTED** [review]

**Review by mlaurent — APPROVED** [review]

File: api/scripts/delete-assessment.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env node
+/* eslint no-console: ["off"] */
+const { Client } = require('pg');
+
+function initialize() {
+  const client = new Client(process.env.DATABASE_URL);
+  client.connect();
+
+  client.query_and_log = function(query) {
```
Comment (DaRiviere) [inline api/scripts/delete-assessment.js]: Extract common code in other module

File: api/tests/unit/script/delete-assessment_test.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        const userEraserWithoutAssessement = new AssessmentEraser(clientStub, queryBuilder, null);
+
+        // when
+        const promise = userEraserWithoutAssessement.delete_dependent_data_from_assessment_id();
+
+        // then
+        return expect(promise).to.be.rejectedWith(Error, 'Missing argument : an assessment id should be provided');
+      });
+
+      it('should delete feedbacks', () => {
```
Comment (DaRiviere) [inline api/tests/unit/script/delete-assessment_test.js]: The following 4 tests to not test that the XXX are deleted but that the query is created. 

Only the 'should exectue all queries' actually tests that the client is called. 

So maybe change all the tests to add a verification that the client is called with the specific query in every test.

File: api/tests/unit/script/delete-assessment_test.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      it('should return the correct query', () => {
+        // given
+        queryBuilderMock.expects('delete_assessment_from_id').once().withArgs(assessment_id);
+
+        // when
+        const promise = subject.delete_assessment_from_id();
+
+        // then
+        return promise.then(() => {
+          queryBuilderMock.verify();
```
Comment (DaRiviere) [inline api/tests/unit/script/delete-assessment_test.js:136]: Maybe the two asserts are redundant, just test that the clientStub is called with the right query string could be enough ?

**Review by DaRiviere — CHANGES_REQUESTED** [review]

File: api/tests/unit/script/delete-assessment_test.js (hunk 3 above)
Comment (Akhilian) [reply api/tests/unit/script/delete-assessment_test.js]: Done :D

File: api/tests/unit/script/delete-assessment_test.js (hunk 4 above)
Comment (Akhilian) [reply api/tests/unit/script/delete-assessment_test.js:136]: Done !

**Review by DaRiviere — APPROVED** [review]

**Pull request merged**